opentok-angular.js

```diff
@@ -311,3 +293,87 @@ ng.module('opentok', [])
       };
     }
   ]);
+
+  const captionSubscriberTracker = (OTSession, $rootScope) => {
+    const MAX_SUBS_ON_SCREEN = 4;
+    const CAPTIONS_TIMEOUT_MSEC = 5 * 1000;
+
+    const captionBox = document.getElementById('caption-render-box');
+
+    let namesByConnectionId = {};
+
+    OTSession.session.on('signal:name', (event) => {
+      namesByConnectionId[event.from.connectionId] = event.data;
+    });
+    // The following two functions are copied from opentok-textchat to ensure the speaker name
+    // and textchat name line up
+    const getNameFromConnection = (connection) => {
+      let id = connection.creationTime.toString();
+      id = id.substring(id.length - 6, id.length - 1);
+      return `Guest${id}`;
+    };
+    const getName = (from) => {
+      if (!namesByConnectionId[from.connectionId]) {
+        namesByConnectionId[from.connectionId] = getNameFromConnection(from);
+      }
+      return namesByConnectionId[from.connectionId];
+    };
+
+    const captionsArray = $rootScope.captionsArray;
+
+    const generateCaptionsString = () => {
+      return captionsArray.map((captionElm) => {
+        return `${captionElm.name}: ${captionElm.captionText}`
+      }).join('\n');
+    }
+    const renderCaptionsArray = () => {
+      const captionString = captionsArray.length > 0 ? generateCaptionsString() : '';
```

Comment:
Nit.  This is a leaky abstraction.  Perhaps this edge case can be moved upstream, to the callee?